clar2wasm/src/wasm_generator.rs

```diff
@@ -563,13 +563,15 @@ impl WasmGenerator {
                 .i32_const(arg_name_len as i32)
                 .global_set(get_global(&self.module, "runtime-error-arg-len")?)
                 .i32_const(ErrorMap::NameAlreadyUsed as i32)
-                .call(self.func_by_name("stdlib.runtime-error"));
+                .call(self.func_by_name("stdlib.runtime-error"))
+                // To avoid having to generate correct return values
+                .unreachable();
+        } else {
+            // Traverse the body of the function
+            self.set_expr_type(body, function_type.returns.clone())?;
+            self.traverse_expr(&mut block, body)?;
```

Comment:
I wonder if we shouldn't traverse the function body even if we don't use it. So that the compiler crashes if we have duplicated argument names AND an error in the body. Or is the typechecking part enough to ensure correctness of the body?

Reply:
Quite true. I think we should have an issue to revisit this part of the code and try to isolate the side effects.